grpcgcp/gcp_balancer.go

```diff
@@ -326,7 +344,22 @@ func (gb *gcpBalancer) UpdateSubConnState(sc balancer.SubConn, scs balancer.SubC
 		delete(gb.scRefs, sc)
 		delete(gb.scStates, sc)
 	}
-	gb.mu.Unlock()
+	if oldS == connectivity.Ready && s != oldS {
+		// Subconn is broken. Remove fallback mapping to this subconn.
+		for k, v := range gb.fallbackMap {
+			if v == sc {
+				delete(gb.fallbackMap, k)
+			}
+		}
+	}
+	if oldS != connectivity.Ready && s == connectivity.Ready {
```

Comment:
This is done to get the least busy subconn when primary subconn fails. Do you want to keep the previous backup for some kind of affinity?

Reply:
No, as it moved out of ready state the affinity is already broken, so we don't need to keep it. We just remove the mapping so that next time an RPC call with key `k` is issued it will get either the originally mapped subconn (if it recovered by that time) or any available least busy subconn.